tests/async.rs

```diff
@@ -93,24 +93,21 @@ where
 
     fn test<C, F>(self, c: C)
     where
-        F: IntoFuture<Item = (), Error = mysql_async::errors::Error>,
+        F: IntoFuture<Item = (), Error = mysql_async::error::Error>,
         C: FnOnce(mysql_async::Conn) -> F,
     {
         let listener = net::TcpListener::bind("127.0.0.1:0").unwrap();
         let port = listener.local_addr().unwrap().port();
+
         let jh = thread::spawn(move || {
             let (s, _) = listener.accept().unwrap();
             MysqlIntermediary::run_on_tcp(self, s)
         });
 
-        let mut core = Core::new().unwrap();
-        let handle = core.handle();
-        let db =
-            core.run(mysql_async::Conn::new(
-                &format!("mysql://127.0.0.1:{}", port),
-                &handle,
-            )).unwrap();
-        core.run(c(db).into_future()).unwrap();
+        let mut runtime = current_thread::Runtime::new().unwrap();
+
+        runtime.block_on(mysql_async::Pool::new(format!("mysql://127.0.0.1:{}", port)).get_conn().and_then(|conn| c(conn))).unwrap();
```

Comment:
In fact, do you even need a `Pool`? Would a plain ol' [`mysql_async::Conn`](https://docs.rs/mysql_async/0.19/mysql_async/struct.Conn.html#method.new) do the trick?

Reply:
will check

Reply:
Interesting, using `mysql_async::Conn` ends up hanging like when I wasn't dropping pool properly in previous iterations.

Reply:
Huh, that is _super_ weird.. Here's a thing you could try:
```rust
tokio::runtime::current_thread::block_on_all(mysql_async::Pool::new(format!("mysql://127.0.0.1:{}", port)).get_conn().and_then(|conn| c(conn))).unwrap();
```
(note that you're no longer creating a `Runtime` -- [`block_on_all`](https://docs.rs/tokio/0.1.19/tokio/runtime/current_thread/fn.block_on_all.html) does that for you)

Reply:
Nice, that one does the trick and is really compact!